shared/common-adapters/markdown/shared.js

```diff
@@ -302,6 +319,30 @@ const rules = {
       return null
     },
   },
+  kbfsPath: {
+    order: SimpleMarkdown.defaultRules.autolink.order - 0.1, // lower than mention
+    match: (source, state, lookBehind) => {
+      const kbfsPathRegex = getKbfsPathRegex(state.markdownMeta)
+      if (!kbfsPathRegex) {
+        return null
+      }
+
+      const matches = SimpleMarkdown.inlineRegex(kbfsPathRegex)(source, state, lookBehind)
+      // Also check that the lookBehind is not text
+      if (matches && (!lookBehind || lookBehind.match(/\B$/))) {
+        return matches
+      }
+      return null
+    },
+    parse: (capture, parse, state) => {
+      const content = pathToString(unescapePath(capture[0]))
```

Comment:
This should probably happen in the component for this instead of here. Then you can just rely on the default implementation of parse and kill this code.